util/prefixer/prefixer.go

```diff
@@ -0,0 +1,50 @@
+package prefixer
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"log/slog"
+	"os"
+	"sync"
+)
+
+type Handler struct {
+	h      slog.Handler
+	m      *sync.Mutex
```

Comment:
Unless I'm reading this wrong, this mutex doesn't appear to be used and should be removed.